### PR TITLE
Finish CLI function implementations

### PR DESCRIPTION
**Pull request by robertbartel**
Complete support for CLI commands to get a list of job ids, to get job state details, and to stop and restart already-running jobs.

## Additions

-  Implementations for _request_job_info_, _request_job_release_, _request_job_restart_, _request_job_status_, _request_job_stop_, and _request_jobs_list_ functions to the _JobClient_ class in the client package
- External request and response message subtypes specifically for the required existing job control and query operations
- _ExistingJobRequestHandler_ external request handler subtype for receiving aforementioned new request types and directing them to scheduler service 
- _JobExecStep_ values `STOPPING` (but not stopped) and `CANCELED` (i.e., first stopped, then resources released) jobs
- Functions to _JobManager_ and implementations to stop and restart jobs
- Functions in _Launcher_ to stop/remove a running job (i.e., the underlying Docker services)
- Function in _JobUtil_ interface to get all job ids (not just running)

## Changes

- Integrating the new external request handler into request service
- Updating scheduler service to handle new job request types
- Updating _Job_ _should_release_resources_ function for both correctness and new status step options
- Updating _JobManager_ _release_resources_ function signature and implementations
- Update Redis-backed job manager to more easily track all jobs
- Adjust Redis-backed job manager to store job id of active jobs instead of the derived Redis key for the job id
- Change _Response_ to inherit from _BasicResponseIndicator_
- Minor adjustments to move away from deprecated attribute assignment when setters are available 
- Updates to tests

## Testing

1. All unit tests pass on development system
2. All "integration" tests pass on primary local development system except for _modeldata_ package; this is expected to be environment-based (related to some current issues with standing up the object store stack) and not due to changes
3. Tests _don't_ all pass (neither set) on a secondary local development system; unsure currently of whether this is due to a user error or an environmental issue of some kind

### CODE REVIEW

File: python/lib/client/dmod/client/request_clients.py
Comment (aaraney) [inline python/lib/client/dmod/client/request_clients.py:8]: Is there a really good reason to use a wildcard import here that im glossing over?

**Review by aaraney — COMMENTED** [review]
Left a few minor comments, but in general this looks good!

File: python/lib/externalrequests/dmod/externalrequests/maas_request_handlers.py
Comment (robertbartel) [reply python/lib/externalrequests/dmod/externalrequests/maas_request_handlers.py:367]: If you are talking about the `_default_required_access_type`, yes, definitely kicking that can down the road for now.

File: python/lib/client/dmod/client/request_clients.py
Comment (robertbartel) [reply python/lib/client/dmod/client/request_clients.py]: Should be taken care of with the previously mentioned change.

File: python/lib/client/dmod/client/request_clients.py
Comment (robertbartel) [reply python/lib/client/dmod/client/request_clients.py]: Should be taken care of with the previously mentioned change.

File: python/lib/client/dmod/client/request_clients.py
Comment (robertbartel) [reply python/lib/client/dmod/client/request_clients.py]: Should be taken care of with the previously mentioned change.

File: python/lib/client/dmod/client/request_clients.py
Comment (robertbartel) [reply python/lib/client/dmod/client/request_clients.py]: Should be taken care of with the previously mentioned change.

File: python/lib/externalrequests/dmod/externalrequests/maas_request_handlers.py
Comment (robertbartel) [reply python/lib/externalrequests/dmod/externalrequests/maas_request_handlers.py:16]: This should also be adjusted now.

File: python/lib/externalrequests/dmod/externalrequests/maas_request_handlers.py
Comment (robertbartel) [reply python/lib/externalrequests/dmod/externalrequests/maas_request_handlers.py:367]: Was I following this correctly, or were you talking about something else?

File: python/lib/externalrequests/dmod/externalrequests/maas_request_handlers.py
Comment (christophertubbs) [inline python/lib/externalrequests/dmod/externalrequests/maas_request_handlers.py:428]: This is most likely a merge issue.

**Review by christophertubbs — CHANGES_REQUESTED** [review]
There's at least one syntax error, so that should be addressed.

File: python/lib/externalrequests/dmod/externalrequests/maas_request_handlers.py
Comment (robertbartel) [reply python/lib/externalrequests/dmod/externalrequests/maas_request_handlers.py]: I think so too.  Just cleaned it up.  Hopefully the tests will pass in a few minutes and this will be good to go.

**Comment by christophertubbs** [comment]
fix the version, brochaho

**Review by aaraney — APPROVED** [review]
This looks good to me. Thanks, @robertbartel. Merge at will

**Comment by robertbartel** [comment]
Despite @aaraney's approval, Github still seems to want me to wait for @christophertubbs, since his review result was to request changes.  I'm going to see what happens if I dismiss that stale review.

**Pull request merged**
